### PR TITLE
Set content-type header to `application/json`

### PR DESCRIPTION
**Pull request by pavelnikolov**
This PR adds `Content-Type: application/json` header to the response of `relay.Handler`'s `ServerHTTP` method. 

I would like to submit more PRs to add some unit tests and improve the way `gqltesting` reports errors. For both of these I would like to add some dependencies like `github.com/stretchr/testify`. Do you have any dependency management tool preferences? I was thinking about `dep` or `glide`, WDYT?

### CODE REVIEW

**Comment by neelance** [comment]
I appreciate that you want to contribute. However, `github.com/stretchr/testify` looks like a repo full of testing stuff that is unnecessary and especially in the Go world frowned upon. Please take a look at this related talk from the dotGo conference: https://www.youtube.com/watch?v=yi5A3cK1LNA

**Comment by pavelnikolov** [comment]
Yeah, it makes sense. I get it why you don't want to introduce big dependences. I wanted to just use one method [`assert.JSONEq`](https://godoc.org/github.com/stretchr/testify/assert#JSONEq) - it gives you a very detailed message showing which fields are different and can save development time. Maybe I should just copy this method to the `gqltesting` package - do you think it is a good idea?
I also wanted to use the testify `mock` package, but I can add unit tests without it - I'll use your example schema and will not mock methods.
You do not have `CONTRIBUTING.md` document and this is why I ask what are your preferences before I spend time writing code, that might be declined later. 

I am using this project and would like to contribute to it as well.

What do you think about the `content-type: application/json` header? We have a 3rd party tool which consumes JSON data it requires the content type header to be present, though. This is why I added it.

File: relay/relay_test.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+	h.ServeHTTP(w, r)
+
+	if w.Code != 200 {
+		t.Errorf("Expected status code 200, got %d.", w.Code)
+	}
+
+	contentType := w.Header().Get("Content-Type")
+	if contentType != "application/json" {
+		t.Errorf("Invalid content-type. Expected [application/json], but isntead got [%s]", contentType)
```
Comment (neelance) [inline relay/relay_test.go]: Typo `isntead`.

File: relay/relay_test.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	}
+
+	expectedResponse := `{"data":{"hero":{"name":"R2-D2"}}}`
+	actualResponse := string(w.Body.Bytes())
+	if expectedResponse != actualResponse {
+		t.Errorf("Invalid response. Expected [%s], but isntead got [%s]", expectedResponse, actualResponse)
```
Comment (neelance) [inline relay/relay_test.go]: Same here.

File: relay/relay_test.go (hunk 1 above)
Comment (pavelnikolov) [inline relay/relay_test.go]: 🤦‍♂️

**Comment by neelance** [comment]
Cool. Thanks for contributing!

File: relay/relay_test.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+		t.Fatalf("Expected status code 200, got %d.", w.Code)
+	}
+
+	contentType := w.Header().Get("Content-Type")
+	if contentType != "application/json" {
+		t.Fatalf("Invalid content-type. Expected [application/json], but instead got [%s]", contentType)
+	}
+
+	expectedResponse := `{"data":{"hero":{"name":"R2-D2"}}}`
+	actualResponse := string(w.Body.Bytes())
```
Comment (shurcooL) [inline relay/relay_test.go]: @pavelnikolov, `string(w.Body.Bytes())` could be simplified to `w.Body.String()`.

Comment (pavelnikolov) [inline relay/relay_test.go]: I'll fix it in the next PR

**Pull request merged**